File: BilligKwhWebApp/Models/TokenModel.cs
using BilligKwhWebApp.Core.Domain;
using System;

namespace BilligKwhWebApp.Models
{
    public class TokenModel
    {
        public int StateCode { get; set; }
        public DateTime RequestedAt { get; set; }
        public DateTime ExpiresAt { get; set; }
        public string AccessToken { get; set; }
        public int UserId { get; set; }
        public int CustomerId { get; set; }

        //The UserId of the user doing the impersonation
        public int ImpersonateFromUserId { get; set; }
        public UserRefreshToken RefreshTokenModel { get; set; }
    }
}

File: BilligKwhWebApp/BilligKwhApp/src/models/user/TokenModel.ts
import { UserRefreshToken } from "./UserRefreshToken";

export class TokenModel {
  stateCode: number;
  requestedAt: string;
  expiresAt: string;
  accessToken: string;
  userId: number;
  customerId: number;
  profileId: number;
  impersonateFromUserId: number;
  refreshTokenModel: UserRefreshToken;
  specificProfile: boolean;
}
